src/Xamarin.Android.Build.Tasks/MSBuild/Xamarin/Android/Microsoft.Android.Sdk.Bindings.Gradle.targets

```diff
@@ -0,0 +1,102 @@
+    </AndroidGradleProjectReference>
+  </ItemDefinitionGroup>
+
+  <Target Name="_GetBuildAndroidGradleProjectsInputs"
+        Condition=" '@(AndroidGradleProjectReference->Count())' != '0' ">
+    <ItemGroup>
+      <_AllGradleFiles Include="$([System.IO.Directory]::GetFiles('%(AndroidGradleProjectReference.FullPath)', '*.*', System.IO.SearchOption.AllDirectories))"
+          Condition= " Exists('%(AndroidGradleProjectReference.FullPath)') "/>
+      <_GradleInputs Include="@(_AllGradleFiles)"
+          Condition="  '%(Extension)' == '.java'
+                    or '%(Extension)' == '.kt'
+                    or '%(Extension)' == '.kts'
+                    or '%(Extension)' == '.gradle'
+                    or '%(Extension)' == '.properties'
+                    or '%(Extension)' == '.xml' " />
+      <_GradleInputs Include="$(MSBuildProjectFullPath)" />
+      <_GradleInputs Include="@(_AndroidMSBuildAllProjects)" />
+    </ItemGroup>
+  </Target>
+
+  <Target Name="BuildAndroidGradleProjects"
+      Condition=" '@(AndroidGradleProjectReference->Count())' != '0' "
+      DependsOnTargets="_SetupMSBuildAllProjects;AndroidPrepareForBuild;_GetBuildAndroidGradleProjectsInputs"
+      Inputs="@(_GradleInputs)"
+      Outputs="$(_BuildAndroidGradleProjectsStamp)" >
+    <!-- Delete outputs folder if inputs have changed to force a partial rebuild -->
+    <RemoveDir Directories="@(AndroidGradleProjectReference->'%(OutputPath)outputs')" />
+
+    <!-- Run assemble task for project outputs, android app and library project outputs are currently supported -->
+    <Gradle ToolPath="%(AndroidGradleProjectReference.FullPath)"
+        Command="assemble%(AndroidGradleProjectReference.Configuration)"
+        ModuleName="%(AndroidGradleProjectReference.ModuleName)"
+        OutputPath="%(AndroidGradleProjectReference.OutputPath)"
+        IntermediateOutputPath="$(MSBuildProjectDirectory)/$(IntermediateOutputPath)"
+        AndroidSdkDirectory="$(AndroidSdkDirectory)"
+        JavaSdkDirectory="$(JavaSdkDirectory)" >
+    </Gradle>
+
+    <ItemGroup>
+      <_AndroidGradleProjectAppOutputs      Include="%(AndroidGradleProjectReference.OutputPath)outputs/**/*.apk" />
+      <_AndroidGradleProjectLibraryOutputs  Include="%(AndroidGradleProjectReference.OutputPath)outputs/**/*.aar"
+          Condition=" '%(ReferenceLibraryOutputs)' == true"
+          Bind="%(Bind)"
+          Pack="%(Pack)"
+          Visible="%(Visible)" />
+    </ItemGroup>
+
+    <AndroidMessage ResourceName="XAGRDLRefLibraryOutputs"
+        FormatArguments="%(_AndroidGradleProjectLibraryOutputs.FullPath)"
+        Condition=" '@(_AndroidGradleProjectLibraryOutputs->Count())' != '0' "
+    />
+
+    <Touch Files="$(_BuildAndroidGradleProjectsStamp)" AlwaysCreate="True" />
+    <ItemGroup>
+      <FileWrites Include="$(_BuildAndroidGradleProjectsStamp);%(AndroidGradleProjectReference.OutputPath)**/*" />
+      <!-- Copy app output files to $(OutputPath) -->
```

Comment:
This comment confuses me, as the subsequent creation of the `@(None)` item group won't *explicitly* copy anything.  Perhaps it will *implicitly* copy them?  I guess?

Reply:
I updated the comment to mention that it is being added to `@(None)`, where `%(CopyToOutputDirectory)` metadata will copy the file implicitly depending on it's value.